From f2523e465fa84e8081ad80d0502b230594fa0177 Mon Sep 17 00:00:00 2001
From: Camelia Groza <camelia.groza@nxp.com>
Date: Mon, 17 Sep 2018 12:39:24 +0300
Subject: [PATCH 595/767] sdk_dpaa: ls1043a errata: align skb_shinfo

Make sure the skb shared info is cache-line aligned when realigning
egress frames.

Signed-off-by: Camelia Groza <camelia.groza@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_sg.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_sg.c b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_sg.c
index f9bb8d92dddb..b66bcb73acab 100644
--- a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_sg.c
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_sg.c
@@ -848,7 +848,7 @@ static struct sk_buff *a010022_realign_skb(struct sk_buff *skb,
 	 *
 	 * Make sure the new linearized buffer will not exceed a page's size.
 	 */
-	nsize = headroom + skb->len +
+	nsize = SMP_CACHE_BYTES + DPA_SKB_SIZE(headroom + skb->len) +
 		SKB_DATA_ALIGN(sizeof(struct skb_shared_info));
 	if (unlikely(nsize > 4096))
 		goto err;
-- 
2.17.0

